internal/service/networkcontainer/data_source_network_container_test.go

```diff
@@ -0,0 +1,152 @@
+package networkcontainer_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+)
+
+var (
+	dataSourceName = "data.mongodbatlas_network_container.test"
+)
+
+func TestAccNetworkContainerDS_basicAWS(t *testing.T) {
+	var (
+		projectName = acctest.RandomWithPrefix("test-acc")
+	)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: dataSourceConfigBasicAWS(projectName, orgID, cidrBlock, providerNameAws, "US_EAST_1"),
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName, &container),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
```

Comment:
all checks should be checking dataSourceName not resource

Reply:
Removed unecessary checks